Do not compile to arm

Hello,

I was hoping to use this library for a project on a raspberry pi but it doesn't compile, it give me this error :

```
cargo build --target=armv7-unknown-linux-gnueabihf
   Compiling interfaces v0.0.1 (file:///home/neodc/tmp/interfaces-rs)
error[E0308]: mismatched types
   --> src/lib.rs:389:45
    |
389 |         let res = unsafe { ioctl(self.sock, SIOCGIFHWADDR, &mut req) };
    |                                             ^^^^^^^^^^^^^ expected u32, found u64
    |
    = help: here are some functions which might fulfill your needs:
            - .count_ones()
            - .count_zeros()
            - .leading_zeros()
            - .trailing_zeros()

error[E0308]: mismatched types
   --> src/lib.rs:400:64
    |
400 |         let addr = unsafe { mem::transmute::<[i8; 6], [u8; 6]>(addr) };
    |                                                                ^^^^ expected i8, found u8

error[E0308]: mismatched types
   --> src/lib.rs:476:45
    |
476 |         let res = unsafe { ioctl(self.sock, SIOCGIFFLAGS, &mut req) };
    |                                             ^^^^^^^^^^^^ expected u32, found u64
    |
    = help: here are some functions which might fulfill your needs:
            - .count_ones()
            - .count_zeros()
            - .leading_zeros()
            - .trailing_zeros()

error[E0308]: mismatched types
   --> src/lib.rs:494:45
    |
494 |         let res = unsafe { ioctl(self.sock, SIOCSIFFLAGS, &mut req) };
    |                                             ^^^^^^^^^^^^ expected u32, found u64
    |
    = help: here are some functions which might fulfill your needs:
            - .count_ones()
            - .count_zeros()
            - .leading_zeros()
            - .trailing_zeros()

error[E0308]: mismatched types
   --> src/lib.rs:520:45
    |
520 |         let res = unsafe { ioctl(self.sock, SIOCGIFMTU, &mut req) };
    |                                             ^^^^^^^^^^ expected u32, found u64
    |
    = help: here are some functions which might fulfill your needs:
            - .count_ones()
            - .count_zeros()
            - .leading_zeros()
            - .trailing_zeros()

error: aborting due to 5 previous errors

error: Could not compile `interfaces`.
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -372,7 +372,7 @@ impl Interface {
     fn hardware_addr_impl(&self) -> Result<HardwareAddr> {
         // We need this IOCTL in order to get the hardware address.
         let SIOCGIFHWADDR = match constants::get_constant("SIOCGIFHWADDR") {
-            Some(c) => c,
+            Some(c) => c as libc::c_ulong,
             None => return Err(InterfacesError::NotSupported("SIOCGIFHWADDR")),
         };
 
@@ -396,8 +396,8 @@ impl Interface {
             addr[i] = req.ifr_hwaddr.sa_data[i];
         }
 
-        // Hardware addresses are `i8`s on Linux for some reason?
-        let addr = unsafe { mem::transmute::<[i8; 6], [u8; 6]>(addr) };
+        // Hardware addresses are `i8`s on some Linux versions for some reason?
+        let addr = unsafe { mem::transmute::<[_; 6], [u8; 6]>(addr) };
         Ok(HardwareAddr(addr))
     }
 
@@ -457,11 +457,11 @@ impl Interface {
     pub fn set_up(&mut self, up: bool) -> Result<()> {
         // We need these IOCTLs in order to get/set the interface flags.
         let SIOCGIFFLAGS = match constants::get_constant("SIOCGIFFLAGS") {
-            Some(c) => c,
+            Some(c) => c as libc::c_ulong,
             None => return Err(InterfacesError::NotSupported("SIOCGIFFLAGS")),
         };
         let SIOCSIFFLAGS = match constants::get_constant("SIOCSIFFLAGS") {
-            Some(c) => c,
+            Some(c) => c as libc::c_ulong,
             None => return Err(InterfacesError::NotSupported("SIOCSIFFLAGS")),
         };
 
@@ -506,7 +506,7 @@ impl Interface {
     #[allow(non_snake_case)]
     pub fn get_mtu(&self) -> Result<u32> {
         let SIOCGIFMTU = match constants::get_constant("SIOCGIFMTU") {
-            Some(c) => c,
+            Some(c) => c as libc::c_ulong,
             None => return Err(InterfacesError::NotSupported("SIOCGIFMTU")),
         };
 

